File: Python3/3191. Minimum Operations to Make Binary Array Elements Equal to One I.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        ans = 0
        N = len(nums)

        for i, n in enumerate(nums):
            if n == 0:
                if i >= N - 2:
                    return -1
                nums[i] = 1
                nums[i + 1] ^= 1
                nums[i + 2] ^= 1
                ans += 1

        return ans

File: Java/3191. Minimum Operations to Make Binary Array Elements Equal to One I.java
class Solution {
    public int minOperations(int[] nums) {
        int N = nums.length, ans = 0;
        for (int i = 0; i < N; i++) {
            if (nums[i] == 0) {
                if (i >= N - 2) {
                    return -1;
                }
                nums[i] = 1;
                nums[i + 1] ^= 1;
                nums[i + 2] ^= 1;
                ans++;
            }
        }
        return ans;
    }
}